### PR TITLE
Adds Subspace Radio

### PR DESCRIPTION
**Pull request by Novacat**
A backpack radio that mechanically works similar to a defibrillator, that can send and receive messages from anywhere regardless of telecoms. Is still susceptible to the usual methods of signal disruption such as jamming.

- Adds the Pathfinder's radio, currently spawns in the Pathfinder's locker.
- Overhauls default channel list, fixing access as well as adding the exploration channel
- Fixes the emergency medical phone and emergency medbay radio link
- Fixes an oversight where access_medical was not properly removed from HoS

### CODE REVIEW

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -474,6 +475,13 @@ var/global/list/default_medbay_channels = list(
 		// we're done here.
 		return 1
 
+//VOREStation Add Start
+	if(bluespace_radio)
+		return Broadcast_Message(connection, M, voicemask, pick(M.speak_emote),
+					  src, message, displayname, jobname, real_name, M.voice_name,
+					  0, signal.data["compression"], list(0), connection.frequency,verb,speaking)
```
Comment (KasparoVy) [inline code/game/objects/items/devices/radio/radio.dm]: Missing a few spaces after commas in the last few parameters here

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -518,7 +526,8 @@ var/global/list/default_medbay_channels = list(
 	if(!(0 in level))
 		var/turf/position = get_turf(src)
 		if(!position || !(position.z in level))
-			return -1
+			if(!bluespace_radio && !(src.bluespace_radio)) //VOREStation Edit
```
Comment (KasparoVy) [inline code/game/objects/items/devices/radio/radio.dm]: Implied `src`?

File: code/game/objects/items/devices/radio/radio_vr.dm [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	toggle_headset()
+
+/obj/item/device/subspaceradio/attack_hand(mob/user)
+	if(loc == user)
+		toggle_headset()
+	else
+		..()
+
+/obj/item/device/subspaceradio/MouseDrop()
+	if(ismob(src.loc))
```
Comment (KasparoVy) [inline code/game/objects/items/devices/radio/radio_vr.dm]: Implied `src`? :thinking:

Is that something that's even reviewed for in Polaris code?

Comment (Novacat) [reply code/game/objects/items/devices/radio/radio_vr.dm]: This is copy-pasted straight from defib code, which is older than Polaris.

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (Novacat) [reply code/game/objects/items/devices/radio/radio.dm]: Yeah, it is the same in the other instances of Broadcast_Message in radio.dm, not sure why that is. I kept it for the sake of consistency.

File: code/game/objects/items/devices/radio/radio_vr.dm (hunk 3 above)
Comment (Arokha) [reply code/game/objects/items/devices/radio/radio_vr.dm]: Defibs are not older than Polaris. Defibs were added to Bay after I added them to Polaris, and Bay never had /tg/ defibs, nor are theirs a port of them.

File: code/game/objects/items/devices/radio/radio.dm (hunk 2 above)
Comment (Arokha) [inline code/game/objects/items/devices/radio/radio.dm]: This doesn't make any sense. This is checking the same thing twice.

File: code/game/objects/items/devices/radio/radio_vr.dm [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	desc = "A powerful new radio recently gifted to Nanotrasen from KHI, this communications device has the ability to send and recieve transmissions from anywhere."
+	icon = 'icons/vore/custom_items_vr.dmi'
+	icon_state = "radiopack"
+	item_state = "parachute"
+	slot_flags = SLOT_BACK
+	force = 5
+	throwforce = 6
+	preserve_item = 1
+	w_class = ITEMSIZE_LARGE
+	action_button_name = "Remove/Replace Headset"
```
Comment (Arokha) [inline code/game/objects/items/devices/radio/radio_vr.dm]: Call it a handset or something

File: code/game/objects/items/devices/radio/radio_vr.dm (hunk 3 above)
Comment (Arokha) [reply code/game/objects/items/devices/radio/radio_vr.dm]: Would rather you remove the explicit srcs unless there's a need for clarity (e.g. src and a higher scope having the same var name), generally bad style.

File: code/game/objects/items/devices/radio/radio_vr.dm [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	if(ismob(headset.loc))
+		var/mob/M = headset.loc
+		if(M.drop_from_inventory(headset, src))
+			to_chat(user, "<span class='notice'>\The [headset] snap back into the main unit.</span>")
+	else
+		headset.forceMove(src)
+
+//Subspace Radio Headset
+/obj/item/device/radio/subspaceradio
+	name = "subspace radio headset"
```
Comment (Arokha) [inline code/game/objects/items/devices/radio/radio_vr.dm]: Handset or something

File: code/game/objects/items/devices/radio/radio.dm [hunk 6]
```diff
@@ -40,6 +40,7 @@ var/global/list/default_medbay_channels = list(
 	var/list/channels = list() //see communications.dm for full list. First channel is a "default" for :h
 	var/subspace_transmission = 0
 	var/adhoc_fallback = FALSE //Falls back to 'radio' mode if subspace not available
+	var/bluespace_radio = FALSE //VOREStation Add - Long Range Radio
```
Comment (Arokha) [inline code/game/objects/items/devices/radio/radio.dm]: Can define this in the _vr file

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (Arokha) [reply code/game/objects/items/devices/radio/radio.dm]: It seems like this should be below the !connection test below, given that you're still using connection. Actually do you know WHY this makes it send even though there's no relay? Is it just the list(0) for the zlevel?

Comment (Novacat) [reply code/game/objects/items/devices/radio/radio.dm]: Yes, it is just the list(0) for the zlevel.

File: code/game/objects/items/devices/radio/radio_vr.dm (hunk 3 above)
Comment (Novacat) [reply code/game/objects/items/devices/radio/radio_vr.dm]: I am not sure how to rewrite it to use less srcs.

**Comment by SpadesNeil** [comment]
Re-review is never coming and original issues were resolved so

**Pull request merged**